File: src/main/java/net/nodus/core/sessionlog/SessionLogService.java
package net.nodus.core.sessionlog;

import java.util.Objects;
import net.nodus.core.site.SiteKeyService;
import net.nodus.database.sessionlog.SessionLog;
import net.nodus.database.sessionlog.SessionLogRepository;
import net.nodus.database.site.entity.SiteKey;
import org.springframework.stereotype.Service;

@Service
public class SessionLogService {

    private final SiteKeyService siteKeyService;
    private final SessionLogRepository sessionLogRepository;

    public SessionLogService(SiteKeyService siteKeyService,
        SessionLogRepository sessionLogRepository) {
        this.siteKeyService = siteKeyService;
        this.sessionLogRepository = sessionLogRepository;
    }

    public SessionLog create(String clientKey, String userSession) {
        SiteKey siteKey = siteKeyService.authenticate(clientKey);

        return sessionLogRepository.save(new SessionLog(
            Objects.requireNonNull(siteKey.getId()),
            siteKey.getSiteId(),
            siteKey.getUserAccountId(),
            userSession
        ));
    }
}

File: src/main/kotlin/net/nodus/sessionlog/SessionLogService.kt
package net.nodus.sessionlog

import net.nodus.site.SiteKeyService
import org.springframework.stereotype.Service

@Service
class SessionLogService(
    private val siteKeyService: SiteKeyService,
    private val sessionLogRepository: SessionLogRepository,
) {
    fun create(clientKey: String, userSession: String): SessionLog {

        val siteKey = siteKeyService.authenticate(clientKey)

        return sessionLogRepository.save(
            SessionLog(
                siteKeyId = requireNotNull(siteKey.id),
                userAccountId = siteKey.userAccountId,
                siteId = siteKey.siteId,
                userSession = userSession,
            )
        )
    }
}